topics/wallet-interaction-standard.md

```diff
@@ -159,7 +188,11 @@ Once the connection between the relying party and the wallet is established, and
 #### Use-Case
 
 1. The relying party sends a `canister_call` request to the wallet.
-2. Upon receiving the request, the wallet validates whether the relying party has the permission to request the action and processes the message following the [ICRC-21](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/consent-msg.md) specification.
+2. Upon receiving the request, the wallet validates wheter it can process the message.
+    - If the request version is not supported by the wallet, the wallet sends a response with an error back to the relying party.
+    - If the network is not supported by the wallet, the wallet sends a response with an error back to the relying party.
+    - If the relying party has not been granted the permission to request the action, the wallet sends a response with an error back to the relying party.
+3. Next, the wallet processes the message following the [ICRC-21](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/consent-msg.md) specification. If the target canister does not support ICRC-21, the wallet should try to decode the arguments by itself and display raw canister call details. If the arguments cannot be decoded, a proper warning must be displayed.
```

Comment:
Imho, a wallet SHOULD already display the warning if the target does not support ICRC-21, and MUST display a warning if the arguments cannot be decoded.